nrf_802154/driver/CMakeLists.txt

```diff
@@ -0,0 +1,101 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+# Platform needs paths to driver headers.
+target_include_directories(nrf-802154-platform
```

Comment:
as mentioned here: https://github.com/nrfconnect/sdk-nrf/pull/4146#discussion_r600302788 `nrf-802154-platform` is a lib defined in Zephyr glue code.
How is a user that tries to integrate directly to `nrfxlib/nrf_802154` aware that this library must be defined ?

Should there be one of following options:
- Check for existence of `nrf-802154-platform` before setting includes
- Create a `nrf-802154-platform` in nrfxlib if not defined externally
- Throw a CMake error if lib is not available

Reply:
actually, what's in principle wrong with this code is the fact that you add `include` and `src` to the platfrom lib.

When using CMake, the principle is to link to the lib from where you need to inherit includes, that is:
```
add_library(nrf-802154-driver)
target_include_directories(nrf-802154-driver
  PUBLIC include # Any depending (linking) on driver need to inherit this, and driver itself needs it.
...
)
```
and then elsewhere do:
```
# Platform needs to link to driver, and inherit public includes.
target_include_directories(nrf-802154-platform PRIVATE nrf-802154-driver)
```

Reply:
The expectation is that the user will always provide the `nrf-802154-platform` target before doing `add_subdirectory(nrfxlib/nrf_802154)`. This fact will be documented in an integration guide in the future.

You are right that it is confusing that some includes are added in a different place than where the target was first created. My initial intention was to prevent the user from supplying the path manually or linking the entire driver target to the platform. I didn't know it was possible to use `target_include_directories` with targets so that's a good tip.

Reply:
>  I didn't know it was possible to use `target_include_directories` with targets so that's a good tip.

sorry, that was a typo, should have been `target_link_libraries()`

Reply:
additional thought while reviewing https://github.com/zephyrproject-rtos/hal_nordic/pull/93.

May `nrf-802154` CMake code should always create the `nrf-802154-platform`.
Then the in integration guide, you may simply write:
> nrfxlib defines a `nrf-802154-platform` library for integrating the `nrf-802154-driver` to the platform.
> It is the responsibility of the project to add required platform sources to this library by calling:
> ```
> target_sources(nrf-802154-platform PRIVATE <platform-sources>)
> ```
>
> It is also possible to add additional compile options or include paths to `nrf-802154-platform` by calling relevant CMake functions.

That way, nrf-802154-driver is responsible for creating the lib, but project is responsible for adding the correct sources implementing the API.

Reply:
> May nrf-802154 CMake code should always create the nrf-802154-platform.
> Then the in integration guide, you may simply write:

I am not totally sure how that would work. If `nrf-802154-platform` were to be defined in `nrf-802154` cmake code, then adding platform sources in https://github.com/zephyrproject-rtos/zephyr/blob/51efef4ff24b04a523bb7da9a11b6484e1248133/modules/hal_nordic/CMakeLists.txt would be impossible, because https://github.com/zephyrproject-rtos/zephyr/blob/51efef4ff24b04a523bb7da9a11b6484e1248133/modules/hal_nordic/CMakeLists.txt#L95 may or may not be executed depending on `CONFIG_NRF_802154_SOURCE_HAL_NORDIC` choice. The `target_sources` doesn't accept not-yet-existent target.

Reply:
I have removed the statements that added include directories to `nrf-802154-platform`.

Reply:
> because https://github.com/zephyrproject-rtos/zephyr/blob/51efef4ff24b04a523bb7da9a11b6484e1248133/modules/hal_nordic/CMakeLists.txt#L95 may or may not be executed depending on `CONFIG_NRF_802154_SOURCE_HAL_NORDIC` choice. The `target_sources` doesn't accept not-yet-existent target.

Correct. So you still need the platform lib, even if not using the 802154 hal nordic sources, I assume.